fractions/microprofile/microprofile-fault-tolerance/src/main/java/org/wildfly/swarm/microprofile/faulttolerance/deployment/RetryContext.java

```diff
@@ -51,23 +51,23 @@ RetryConfig getConfig() {
      *
      * @param retryContext
      * @param throwable
-     * @return {@code true} if we should try again or rethrows the specified exception
+     * @return an exception to rethrow or null if we should try again
      */
-    boolean nextRetry(Throwable throwable) throws Exception {
+    Exception nextRetry(Throwable throwable) {
         // Decrement the retry count for this attempt
         remainingAttempts.decrementAndGet();
         // Check the exception type
         if (shouldRetryOn(throwable, System.nanoTime())) {
             delayIfNeeded();
-            return true;
+            return null;
         } else {
             if (throwable instanceof Error) {
                 throw (Error) throwable;
```

Comment:
Should `return`?

Reply:
Yes, it should return. We've changed the method return type (see also javadoc ;-) and only throw `Error`s.

Reply:
No, I understand that the exceptions are to be `return`-ed now, was just wondering if this particular one should really be `throw`-n. It seems it should.